File: 1796-second-largest-digit-in-a-string/1796-second-largest-digit-in-a-string.js
/**1796. Second Largest Digit in a String
 * @param {string} s
 * @return {number}
 */
var secondHighest = function(s) {
    let max = -1;
    let scMax = -1;
    for(let i  = 0;i<s.length;i++){

        if(s[i] >= '0' && s[i] <= '9'){
            if(s[i] > max){
                 scMax = max;
                 max = Number(s[i]);
            }else{
                if(s[i] > scMax && s[i] < max){
                    scMax = Number(s[i]);
                }
            }
        }
    }
    return scMax;
};

File: 1796-second-largest-digit-in-a-string/1796-second-largest-digit-in-a-string.java
class Solution {
    public int secondHighest(String s) {
        int max = -1;
        int second_max = -1;

         for(int i = 0;i<s.length();i++){
             if (s.charAt(i) >= '0' && s.charAt(i) <= '9'){
                 int digit = s.charAt(i) - '0';
                 if(max < digit) {
                     second_max = max ;
                     max = digit;
                 }
                 else if (digit < max && second_max < digit){
                     second_max = digit;
                 }
             }
         }
        return second_max;
    }
}